src/Page/Food/Builder.elm

```diff
@@ -637,7 +699,52 @@ updateIngredientFormView { excluded, db, ingredient, impact, transportImpact } =
             , transportImpact
             ]
         ]
-    ]
+
+
+type alias BonusViewConfig msg =
+    { bonusImpact : Unit.Impact
+    , bonusSplit : Split
+    , domId : String
+    , name : String
+    , selectedImpact : Impact.Definition
+    , updateEvent : Split -> msg
+    }
+
+
+ingredientBonusView : BonusViewConfig Msg -> Html Msg
+ingredientBonusView { name, bonusImpact, bonusSplit, domId, selectedImpact, updateEvent } =
+    div [ class "IngredientBonus" ]
+        [ label [ for domId, class "BonusName text-nowrap text-muted" ] [ text name ]
+        , input
+            [ type_ "range"
+            , id domId
+            , class "BonusRange form-range"
+            , Attr.min "0"
+            , Attr.max "100"
+            , step "1"
+            , Attr.value <| Split.toPercentString bonusSplit
+            , onInput
+                (String.toInt
+                    >> Maybe.map (Split.fromPercent >> Result.withDefault Split.zero)
+                    >> Maybe.withDefault Split.zero
```

Comment:
Nit: would this work to avoid repeating `Split.zero`?
```
String.toInt
    >> Maybe.andThen (Split.fromPercent >> Result.toMaybe)
    >> Maybe.withDefault Split.zero
```